### ISSUE TITLE
Compiler error: Stack too deep. Strange behaviour.

### ISSUE THREAD

**Issue by Georgi87** [ethereum/cpp-ethereum#2478]
When I use the following contract and comment in the line //SS[a].x; the compiler fails.
I understand that the getter for the storage is like a function call and the parameters (a) is added to the stack. But why does the compiler fail? At this step there are still 2 stack slots available because only 14 have been used for function parameters incl return parameters.

```
contract TestContract {

    struct S {
      uint x;
    }
    mapping (uint => S) SS;

    function t1(uint a, uint b, uint c, uint d, uint e, uint f, uint g, uint h, uint i, uint j, uint k, uint l, uint m) returns (uint) {
        // 14 local vars used for function parameters (13) and return parameter (1)
        //SS[a].x; //When I run this line of code I get the "Compiler error: Stack too deep, try removing local variables." Why does it need more than 2 local vars, which could be released afterwards anyways.
        return t2(a, b);
        // function call parameters (2) add 2 local vars
    }

    function t2(uint a, uint b) returns (uint) {
        return 2;
    }

}
```

**Comment by chriseth** [comment]
It turns out that there was a check for the stack height that was a bit too restrictive.
Thanks for reporting (and checking in such detail)!

**Comment by Georgi87** [comment]
Thanks chriseth for confirming! Look forward to the fix.

**Comment by chriseth** [comment]
Fixed by #2480.
